apps/api/src/app/user/update-user-setting.dto.ts

```diff
@@ -88,6 +89,9 @@ export class UpdateUserSettingDto {
   @IsOptional()
   locale?: string;
 
+  @IsOptional()
+  portfolioSettings?: PortfolioReportRule['settings'];
```

Comment:
Attention: data needs to be stored on `xRayRules` to make it work

You could adapt `RuleSettings` in `XRayRulesSettings` to something like this:

```
interface RuleSettings extends Pick<PortfolioReportRule, 'settings'> {
  isActive: boolean;
}
```